controller/api/destination/watcher/cluster_store.go

```diff
@@ -68,32 +70,50 @@ const (
 // When created, a pair of event handlers are registered for the local cluster's
 // Secret informer. The event handlers are responsible for driving the discovery
 // of remote clusters and their configuration
-func NewClusterStore(k8sAPI *k8s.API, enableEndpointSlices bool) (*ClusterStore, error) {
-	return newClusterStoreWithDecoder(k8sAPI, enableEndpointSlices, decodeK8sConfigFromSecret)
+func NewClusterStore(client kubernetes.Interface, namespace string, enableEndpointSlices bool) (*ClusterStore, error) {
+	return NewClusterStoreWithDecoder(client, namespace, enableEndpointSlices, decodeK8sConfigFromSecret)
+}
+
+func (cs *ClusterStore) Sync(stopCh <-chan struct{}) {
+	go func() {
+		cs.secrets.Run(stopCh)
+	}()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+
+	cs.log.Infof("waiting for cluster store to sync")
+	if !cache.WaitForCacheSync(ctx.Done(), cs.secrets.HasSynced) {
+		cs.log.Fatal("failed to sync cluster store")
+	}
+	cs.log.Infof("cluster store synced")
 }
 
 // newClusterStoreWithDecoder is a helper function that allows the creation of a
 // store with an arbitrary `configDecoder` function.
-func newClusterStoreWithDecoder(k8sAPI *k8s.API, enableEndpointSlices bool, decodeFn configDecoder) (*ClusterStore, error) {
+func NewClusterStoreWithDecoder(client kubernetes.Interface, namespace string, enableEndpointSlices bool, decodeFn configDecoder) (*ClusterStore, error) {
+	secretsInformerFactory := informers.NewSharedInformerFactoryWithOptions(client, k8s.ResyncTime, informers.WithNamespace(namespace))
+	secrets := secretsInformerFactory.Core().V1().Secrets().Informer()
```

Comment:
I considered that, but it would mean also passing in a dynamic client and a l5dCrd client which never get used.  It seemed simpler to avoid all of that machinery and just build what we need.

Reply:
Actually those two clients can be passed as nils.